File: Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0;
        int j = height.size() - 1;
        int V = min(height[i], height[j]) * (j-i);

        while(i!=j){
            if(height[i]<height[j]){
                i++;
                if(height[i]>height[i-1] && i!=j){
                    V = max(V , min(height[i], height[j]) * (j-i));
                }
            }
            else{
                j--;
                if(height[j]>height[j+1] && i!=j){
                    V = max(V , min(height[i], height[j]) * (j-i));
                }
            }
        }
        return V;
    }
};

File: Container With Most Water/Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        start =0 
        end=len(height)-1
        area=0
        while end>=start:
            a=(end-start)*min(height[start],height[end])
            if height[end]<height[start]:
                end=end-1
            elif height[end]>=height[start]:
                start=start+1
            if a>area:
                area=a
        return area
